src/dodal/beamlines/i22.py

```diff
@@ -0,0 +1,80 @@
+from ophyd_async.panda import PandA
+
+from dodal.beamlines.beamline_utils import device_instantiation, get_directory_provider
+from dodal.beamlines.beamline_utils import set_beamline as set_utils_beamline
+from dodal.devices.i22.pilatus import I22HDFStatsPilatus
+from dodal.devices.linkam3 import Linkam3
+from dodal.devices.tetramm import TetrammDetector
+from dodal.log import set_beamline as set_log_beamline
+from dodal.utils import get_beamline_name
+
+BL = get_beamline_name("i22")
+set_log_beamline(BL)
+set_utils_beamline(BL)
+
+
+def saxs(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> I22HDFStatsPilatus:
+    return device_instantiation(
+        I22HDFStatsPilatus,
+        "saxs",
+        "-EA-PILAT-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
+    )
+
+
+def waxs(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> I22HDFStatsPilatus:
+    return device_instantiation(
+        I22HDFStatsPilatus,
+        "waxs",
+        "-EA-PILAT-03:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        directory_provider=get_directory_provider(),
+    )
+
+
+def panda(wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False) -> PandA:
+    return device_instantiation(
+        PandA, "panda-01", "-EA-PANDA-01:", wait_for_connection, fake_with_ophyd_sim
+    )
+
+
+def linkam(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Linkam3:
+    return device_instantiation(
+        Linkam3, "linkam", "-EA-TEMPC-05", wait_for_connection, fake_with_ophyd_sim
+    )
+
+
+def tetramm1(
```

Comment:
Perhaps this is a slightly wider discussion, but should we be naming the _device_ or the _scientific function_ of the device in dodal?

For the detectors we've called them `waxs` and `saxs` - i.e. the scientific function of the device (as opposed to, say, `pilatus1` and `pilatus2`).

So should these tetramms be `incident_beam_monitor` and `transmitted_beam_monitor`, or `user_tetramm` for example? (I'm assuming that's what `i0` and `it` mean, but could be wrong...)

Reply:
There's definitely a discussion to be had: I think it comes down to beamline scientist/support engineer preference on "how easy is this to type" vs. "how completely does this describe the function of the device" vs. "how completely does this describe the form of the device" (I think "how easy to type" is going to get a lot of traction...).

Do you have preference? I think `incident_beam` and `transmitted_beam`, with sufficient docstrings on the type and that it doesn't implement Movable makes it clear that it is a Monitor/Sensor.

Reply:
Sure to be honest I don't have a particularly strong opinion on the actual chosen names, it just jumped out to me as  inconsistent with the saxs/waxs detectors. Happy for you to spin this off into an issue or discussion or strawpoll on slack or something if you prefer...

Reply:
Made an issue, but also renamed for i22 to something I think is sensible. Will return to p38 when can discuss with p38 interested parties.

Reply:
For your info I asked i22 scientists about what naming they'd generally prefer in a meeting this morning - their general preference is for scientific meaning rather than device name. E.g. they suggested `i0` and `it` are preferred over `tetramm1` etc.